lib/sidekiq/cron/job.rb

```diff
@@ -638,41 +662,82 @@ def not_past_scheduled_time?(current_time)
         true
       end
 
-      # Redis key for set of all cron jobs.
-      def self.jobs_key
-        "cron_jobs"
+      def self.default_if_blank(namespace)
+        if namespace.nil? || namespace == ''
+          Sidekiq::Cron.configuration.default_namespace
+        else
+          namespace
+        end
+      end
+
+      def self.job_keys_from_namespace(namespace = Sidekiq::Cron.configuration.default_namespace)
+        Sidekiq.redis do |conn|
+          if namespace == '*'
+            namespaces = conn.keys(jobs_key(namespace))
+            namespaces.collect { |name| conn.smembers(name) }.flatten
+          else
+            conn.smembers(jobs_key(namespace))
+          end
+        end
+      end
+
+      def self.default_or_no_namespace?(namespace)
+        namespace.nil? || namespace == '' || namespace == '*' ||
+          namespace == Sidekiq::Cron.configuration.default_namespace
+      end
+
+      def self.migrate_old_jobs_if_needed!
+        Sidekiq.redis do |conn|
+          old_job_keys = conn.smembers('cron_jobs')
+          old_job_keys.each do |old_job|
+            old_job_hash = conn.hgetall(old_job)
+            old_job_hash[:namespace] = Sidekiq::Cron.configuration.default_namespace
+            create(old_job_hash)
+          end
+
+          old_job_keys.each { |old_job_key| conn.srem('cron_jobs', old_job_key) }
```

Comment:
why do we need to loop the old_job_keys again here? why not just call the `conn.srem` in the first loop after migrated job has been created?